sql/hive/src/main/scala/org/apache/spark/sql/hive/hiveWriterContainers.scala

```diff
@@ -212,9 +213,14 @@ private[spark] class SparkHiveDynamicPartitionWriterContainer(
       .zip(row.toSeq.takeRight(dynamicPartColNames.length))
       .map { case (col, rawVal) =>
         val string = if (rawVal == null) null else String.valueOf(rawVal)
-        s"/$col=${if (string == null || string.isEmpty) defaultPartName else string}"
-      }
-      .mkString
+        s"/$col=${
+          if (string == null || string.isEmpty) {
```

Comment:
Can you move this computation out of string interpolation into a variable.  Its kinda odd to have a multi-line string thats not really multi line.